submissions/ik-web/friends-app/js/script.js

```diff
@@ -0,0 +1,132 @@
+document.addEventListener('DOMContentLoaded', () => {
+
+    const url = 'https://randomuser.me/api/?results=80&seed=fa21';
+    const usersBox = document.querySelector('.usersBox');
+
+    // ---------------------------------------------------------------
+    function createUserCard(name, photo, gender, age, phone, email) {
+        return `
+            <div class="userCard">
+                <div class="userCard__inner">
+                    <h2 class="userCard__userName">${name}</h2>
+                    <div class="userCard__userPhoto"><img src=${photo} alt="photo"></div>
+                    <div class="userCard__userInfo userInfo">
+                        <div class="userInfo__item">${gender}</div>
+                        <div class="userInfo__item">${age} years old</div>
+                        <div class="userInfo__item">
+                            <a href="tel:${phone}">${phone}</a>
+                        </div>
+                        <div class="userInfo__item">
+                            <a href="mailto:${email}">${email}</a>
+                        </div>                    
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+    
+    function getUsersInfo() {
+        return fetch(url)
+        .then(response => response.json())
+        .then(response => response.results)
+        .catch(err => window.location.reload());
```

Comment:
Why you need reload in case of error?

Reply:
To re-request data from the server.  Now, it seems to me that this is bad practice ... It is better to display a message like "Something went wrong... Please reload this page" ?

Reply:
Nope. Don't make users solve your problems. If you believe that reloading of data will help - you should do it by yourself.

Reply:
In the current task, in my opinion, this is the correct solution, what do you think about this? can I leave it as it is?

Reply:
Can I hear reasons why you think this is a good solution?

Reply:
In this case, I see the problem the server doesn't always return the requested data, but when the page reloads, the problem is solved. As you sayd above, the user doesn't have to solve our problems, and it would be bad to ask him to reload the page if it is possible to do an automatic reload in case of this error.

Reply:
Yes, but I meant that you can send a request, wait for a result, if there is an empty response or error - fetch data again.

From the user's perspective loading process will take a slightly bigger time. Even if he will mention this difference

Reply:
I can't understand how I need to write this... It should be something like a loop request, recursion?

Reply:
Yeap. It can be async loop or recursion.

Other students already implemented this logic, you can find it in closed PRs

Reply:
Yeap. It can be async loop or recursion.

Other students already implemented this logic, you can find it in closed PRs

Reply:
Thanks a lot! I did it by recursion